src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml

```diff
@@ -186,6 +186,12 @@ services:
       - '@prestashop.core.grid.query.doctrine_query_parser'
       - 'profile'
 
+  prestashop.core.grid.data_factory.profiles_decorator:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Data\Factory\ProfileGridDataFactoryDecorator'
+    arguments:
+      - '@prestashop.core.grid.data_factory.profiles'
+      - '@security.helper'
+
```

Comment:
Hi @0x346e3730, can you plz feat with https://github.com/PrestaShop/ADR/blob/master/0019-FQCN-Autowiring.md

1. create the FQCN version of '@prestashop.core.grid.data_factory.profiles' and '@security.helper' and also aliases version (for BC compatibility)
2. Declare the new service like this:
```php
PrestaShop\PrestaShop\Core\Grid\Data\Factory\ProfileGridDataFactoryDecorato:
    autowire: true
    public: false
```

Reply:
Done